docs/guide-disappearance.md

```diff
@@ -3,93 +3,114 @@ id: guide-disappearance
 title: Appearance and Disappearance
 ---
 
-Sometimes you need to test that an element is present and then disappears or
-vice versa.
+Sometimes you need to test that an element is present and then disappears or the
+other way around.
 
 ## Waiting for appearance
 
-If you need to wait for an element to appear, the [async wait
-utilities][async-api] allow you to wait for an assertion to be satisfied before
-proceeding. The wait utilities retry until the query passes or times out.
+If you need to wait for an element to appear, [`findBy`][find-by] and the [async
+wait utility][async-api] `waitFor` allow you to wait for an assertion to be
+satisfied before proceeding. The wait utilities retry until the query passes or
+times out.
 
-```jsx
+Wait for appearance and return the element using `findBy`:
+
+```javascript
+test('movie title appears', async () => {
+  // element is initially not present...
+
+  const movieTitle = await screen.findByText('the lion king')
+  expect(movieTitle).toBeInTheDocument()
+})
+```
+
+Or wait for appearance using `waitFor` and `getBy`:
+
+```javascript
 test('movie title appears', async () => {
   // element is initially not present...
 
-  // wait for appearance
   await waitFor(() => {
-    expect(getByText('the lion king')).toBeInTheDocument()
+    expect(screen.getByText('the lion king')).toBeInTheDocument()
   })
-
-  // wait for appearance and return the element
-  const movie = await findByText('the lion king')
 })
 ```
 
+These do more or less the same thing (`findBy` uses `waitFor` under the hood),
+but the `findBy` version results in simpler code and a better error message.
+
+_Note: `toBeInTheDocument()` comes from [`@testing-library/jest-dom`][jest-dom]_
+
 ## Waiting for disappearance
 
-The `waitForElementToBeRemoved` [async helper][async-api] function uses a
-callback to query for the element on each DOM mutation and resolves to `true`
-when the element is removed.
+When you want to wait until an element has disappeared, you can use
+`waitForElementToBeRemoved`.
+
+The `waitForElementToBeRemoved` [async wait utility][async-api] takes an element
+as an argument and waits until that element has been removed from the document.
+
+```javascript
+test('movie title no longer present in document', async () => {
+  // element is present
+  const movie = screen.queryByText('the mummy')
+  await waitForElementToBeRemoved(movie)
```

Comment:
I see @kentcdodds commented over here that this should receive a callback, not the element directly: https://github.com/testing-library/react-testing-library/issues/673#issuecomment-643706909

I feel like I've seen that doing it this way works too though, so I'm not sure.

Reply:
It can either receive a callback, an element, or an array of elements:

https://github.com/testing-library/dom-testing-library/blob/master/types/__tests__/type-tests.ts#L158-L164

Reply:
might be worth expanding on this section later with a note about working with React, because sometimes the node gets destroyed and replaced, making these types of tests either pass or fail when they shouldn't